framework/src/csg/CSGYCylinder.C

```diff
@@ -0,0 +1,37 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CSGYCylinder.h"
+
+namespace CSG
+{
+
+CSGYCylinder::CSGYCylinder(
+    const std::string & name, const Real x0, const Real z0, const Real r, std::string boundary)
+  : CSGSurface(name, MooseUtils::prettyCppType<CSGYCylinder>(), boundary), _x0(x0), _z0(z0), _r(r)
+{
+}
+
+std::unordered_map<std::string, Real>
+CSGYCylinder::getCoeffs() const
+{
+  std::unordered_map<std::string, Real> coeffs = {{"x0", _x0}, {"z0", _z0}, {"r", _r}};
+  return coeffs;
+}
+
+Real
+CSGYCylinder::evaluateSurfaceEquationAtPoint(const Point & p) const
```

Comment:
I suspect we are going to need a "fuzzy" evluate with a tolerance. maybe it should be the only API?
```
evaluateSurfaceEquationAtPoint(const Point & p, Real tol = 0)
``` 
for example?

Reply:
probably not the only API, looking at what we have in MooseUtils. 
I m happy with this one.